### DISCUSSION TITLE
Design & implement sandboxing for Zeek processes

### DISCUSSION THREAD

**Discussion by rsmmr** [zeek/zeek#852] [Ideas]
There are various ways to avoid running Zeek as root, including `setcap` and sandboxing through external tools (jails, systemd). However, with the new supervisor architecture we could now do more inside Zeek as well by sandboxing processes as we spawn them. As that's however OS-specific stuff, I propose we extend our plugin API so that such functionality can be provided externally; and then create a corresponding Linux plugin to implement it.

For reference, see the earlier discussion on this PR: #754.

**Comment by GTrunSec** [comment]
https://github.com/zeek/zeek/issues/939

NixOS: Deploy Zeek with zeekctl
https://github.com/NixOS/nixpkgs/pull/86494#issuecomment-623093567 

if we did not solve the Zeekctl problem which running on NixOS(sandbox mode). I'm going to write a Systemd of Zeek for NixOS that can be running with some arguments of the interface instead of zeekctl to deploy it.

**Comment by rsmmr** [comment]
There's a systemd sandboxing example here: https://github.com/zeek/zeek/pull/754#issuecomment-580652275
